### ISSUE TITLE
Bug in .cif writing fractional XYZ

### ISSUE THREAD

**Issue by cdvonbargen** [openbabel/openbabel#2160]
The cif writer doesn't guarantee a space between values in the loop_ block as currently implemented, which can cause a malformed block:

Specifically, the write operation here:
https://github.com/openbabel/openbabel/blob/master/src/formats/cifformat.cpp#L1651

can create blocks like the following, given a sufficient number of digits in fractional x/y/z:
```
loop_
    _atom_site_label
    _atom_site_type_symbol
    _atom_site_fract_x
    _atom_site_fract_y
    _atom_site_fract_z
    _atom_site_occupancy
    H0      H    -71.99400-128.76240  56.30360   1.000
    C1      C    -68.82210-127.92280  55.65700   1.000
    ...
```

**Comment by ghutchis** [comment]
I have no idea how -71.9, -128.76, etc. are fractional coordinates near zero. While I take your point, I'd really want to see a real example where this happens. IMHO such coordinates need to be moved to the primitive unit cell or something close to it.

Can you provide a specific example?

**Comment by cdvonbargen** [comment]
Attached the problematic sdf file I was working with, though I do not know its origin.

[kevlar.sdf.zip](https://github.com/openbabel/openbabel/files/4452184/kevlar.sdf.zip)

**Comment by patrickfuller** [comment]
Agreed with @ghutchis. We could mod this to force some spacing but I don't see those large numbers as fractional coordinates.

Untested, but something like

```
%-8s%-5s%.5f%10.5f%10.5f%8.3f\n. # old
%-8s %-4s %.5f %9.5f %9.5f %7.3f\n. # new?
```

Regarding why this is fixed format - it's because there are many parsers that don't follow the CIF spec. Loosening the format here could break a lot of grad students' code.

I'll caveat with I've been out of grad school for a while now and the environment may have changed.

**Comment by afonari** [comment]
Thanks for the reply @patrickfuller. By looking closer, I have a question, why the format of X is not the same as for Y and Z:`%.5f` vs. `%9.5f %9.5f` ?

**Comment by patrickfuller** [comment]
Honestly, I don't remember. Might have been trying to be back-compatible with the old code, maybe something with left vs. right aligning the spaces, and maybe just a type. It's been a while!

**Comment by afonari** [comment]
Thanks. Created https://github.com/openbabel/openbabel/pull/2170. It is not fixing a bigger problem probably of CIF missing the cell information.

**Comment by patrickfuller** [comment]
If you're not getting fractional coordinates in [0, 1], then the structure itself is malformed. Or you're trying to use the CIF format for something that's non-crystalline, which is not its intended purpose.

Maybe the right answer here is for openbabel to error instead of allowing it to pass through?

**Comment by ghutchis** [comment]
For now, this is closed by #2170 - although I agree with @patrickfuller that we should throw a warning if there isn't a unit cell.
